### ISSUE TITLE
Can the HttpVerb enum be expanded with Options?

### ISSUE THREAD

**Issue by 4l1ce** [pact-foundation/pact-net#126]
can this enum be expanded with options http verb?
`public enum HttpVerb
  {
    NotSet,
    Get,
    Post,
    Put,
    Delete,
    Head,
    Patch,
  }`

Thank you

**Comment by 4l1ce** [comment]
I'd be happy to create a pull request, but I don't have enough permissions.

**Comment by neilcampbell** [comment]
Yep sure can.

**Comment by neilcampbell** [comment]
Hey @bethesque does the ruby standalone have support for the Options HTTP method? I am getting the below when trying to use it.

The Events API request for OPTIONS http://localhost:9222/test failed. Response Status: 500, Response Body: {"message":"Error ocurred in mock service: NoMethodError - undefined method `match' for nil:NilClass","backtrace":["C:/src/OS/pact-net/Samples/EventApi/Consumer.Tests/bin/Release/net46/pact-win32/lib/vendor/ruby/2.2.0/gems/pact-mock_service-2.6.0/lib/pact/mock_service/request_handlers/options.rb:35:in `is_administration_request?'","C:/src/OS/pact-net/Samples/EventApi/Consumer.Tests/bin/Release/net46/pact-win32/lib/vendor/ruby/2.2.0/gems/pact-mock_service-2.6.0/lib/pact/mock_service/request_handlers/options.rb:17:in `match?'","C:/src/OS/pact-net/Samples/EventApi/Consumer.Tests/bin/Release/net46/pact-win32/lib/vendor/ruby/2.2.0/gems/pact-mock_service-2.6.0/lib/pact/mock_service/request_handlers/base_request_handler.rb:17:in `call'","C:/src/OS/pact-net/Samples/EventApi/Consumer.Tests/bin/Release/net46/pact-win32/lib/vendor/ruby/2.2.0/gems/rack-2.0.3/lib/rack/cascade.rb:33:in `block in call'","C:/src/OS/pact-net/Samples/EventApi/Consumer.Tests/bin/Release/net46/pact-win32/lib/vendor/ruby/2.2.0/gems/rack-2.0.3/lib/rack/cascade.rb:24:in `each'","C:/src/OS/pact-net/Samples/EventApi/Consumer.Tests/bin/Release/net46/pact-win32/lib/vendor/ruby/2.2.0/gems/rack-2.0.3/lib/rack/cascade.rb:24:in `call'","C:/src/OS/pact-net/Samples/EventApi/Consumer.Tests/bin/Release/net46/pact-win32/lib/vendor/ruby/2.2.0/gems/pact-mock_service-2.6.0/lib/pact/consumer/mock_service/cors_origin_header_middleware.rb:11:in `call'","C:/src/OS/pact-net/Samples/EventApi/Consumer.Tests/bin/Release/net46/pact-win32/lib/vendor/ruby/2.2.0/gems/pact-mock_service-2.6.0/lib/pact/consumer/mock_service/error_handler.rb:13:in `call'","C:/src/OS/pact-net/Samples/EventApi/Consumer.Tests/bin/Release/net46/pact-win32/lib/vendor/ruby/2.2.0/gems/pact-mock_service-2.6.0/lib/pact/mock_service/app.rb:33:in `call'","C:/src/OS/pact-net/Samples/EventApi/Consumer.Tests/bin/Release/net46/pact-win32/lib/vendor/ruby/2.2.0/gems/pact-mock_service-2.6.0/lib/pact/consumer/mock_service/set_location.rb:14:in `call'","C:/src/OS/pact-net/Samples/EventApi/Consumer.Tests/bin/Release/net46/pact-win32/lib/vendor/ruby/2.2.0/gems/rack-2.0.3/lib/rack/handler/webrick.rb:86:in `service'","C:/src/OS/pact-net/Samples/EventApi/Consumer.Tests/bin/Release/net46/pact-win32/lib/vendor/ruby/2.2.0/gems/webrick-1.3.1/lib/webrick/httpserver.rb:138:in `service'","C:/src/OS/pact-net/Samples/EventApi/Consumer.Tests/bin/Release/net46/pact-win32/lib/vendor/ruby/2.2.0/gems/webrick-1.3.1/lib/webrick/httpserver.rb:94:in `run'","C:/src/OS/pact-net/Samples/EventApi/Consumer.Tests/bin/Release/net46/pact-win32/lib/vendor/ruby/2.2.0/gems/webrick-1.3.1/lib/webrick/server.rb:191:in `block in start_thread'"]}

**Comment by bethesque** [comment]
Yes, CORS Is supported, see the docs here: https://github.com/pact-foundation/pact-mock_service/wiki/Using-the-mock-service-with-CORS

```ruby
        def is_administration_request? env
          env["HTTP_ACCESS_CONTROL_REQUEST_HEADERS"].match(/x-pact-mock-service/i)
        end
```

It's blowing up because there is no `Access-Control-Request-Headers` header. The Ruby code should be able handle this better, but at the same time, I think it's a required header.

**Comment by neilcampbell** [comment]
@4l1ce Just wanted to confirm that you are using the OPTIONS HTTP method for CORS related things and not something else?

**Comment by 4l1ce** [comment]
@neilcampbell yes, UI is making call using OPTIONS then regular GETs. Adding the header allow-cross-origin is not enough :)

**Comment by neilcampbell** [comment]
@4l1ce Updated. This is available in 2.1.2. As Beth mentioned you will need to ensure the "Access-Control-Request-Headers" header is sent in the request.
